stores/cachestore/cache_store.go

```diff
@@ -0,0 +1,49 @@
+package cachestore
+
+import (
+	"os"
+	"time"
+
+	"github.com/mailchain/mailchain/encoding"
+	"github.com/mailchain/mailchain/internal/hash"
+
+	"github.com/spf13/afero"
+)
+
+type CacheStore struct {
+	cache afero.Fs
+}
+
+func NewCacheStore(cacheTimeout time.Duration) *CacheStore {
+	base := afero.NewOsFs()
+	layer := afero.NewMemMapFs()
+	cache := afero.NewCacheOnReadFs(base, layer, cacheTimeout)
```

Comment:
I didnt think of the memory cache as getting from local disk is already an improvement over a network. This is a nice touch though.